diff-pdf.cpp

```diff
@@ -170,91 +245,97 @@ cairo_surface_t *diff_images(int page, cairo_surface_t *s1, cairo_surface_t *s2,
     // and the other two channels from s2:
 
     // first, copy s1 over:
-    if ( s1 )
-    {
-        unsigned char *out = datadiff + r1.y * stridediff + r1.x * 4;
-        for ( int y = 0;
-              y < r1.height;
-              y++, data1 += stride1, out += stridediff )
-        {
-            memcpy(out, data1, r1.width * 4);
-        }
-    }
+	if (s1) {
+		unsigned char *out = datadiff + (r1.y - rdiff.y) * stridediff + (r1.x - rdiff.x) * 4;
+		for (int y = 0; y < r1.height; y++, data1 += stride1, out += stridediff) {
+			memcpy(out, data1, r1.width * 4);
+		}
+	}
```

Comment:
Please look at this snippet as representative of the same changes all over the PR. It

- changes whitespace (probably to TABs which aren't used anywhere in the file)
- it radically reformats the code block pointlessly
- it doesn't respect the file's existing formatting (w.r.t. `{` in particular)

Please avoid making such changes and please avoid TABs. 

P.S. You can use `git push —force` to replace commits with new ones.